### PR TITLE
Fetch and configure the engine to start on a specific network

### PR DESCRIPTION
**Pull request by antho1404**
The command `daemon:start` now accepts the flag `network` which has by default the `mesg-dev-chain` network.

When starting the daemon it will fetch the configurations and write them in the right directory.

The directory is now `.mesg/{{network}}/` instead of `.mesg/` directly, this allows switching to different networks without losing all the data.

### CODE REVIEW

File: packages/cli/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,9 @@
 
 #### Breaking Changes
 #### Improvements
+
+- [](https://github.com/mesg-foundation/js-sdk/pull/) Fetch and configure the engine to start on a specific network.
```
Comment (antho1404) [inline packages/cli/CHANGELOG.md]:
```suggestion
- [#168](https://github.com/mesg-foundation/js-sdk/pull/168) Fetch and configure the engine to start on a specific network.
```

File: packages/cli/src/commands/daemon/start.ts [hunk 2]
```diff
@@ -83,4 +92,26 @@ export default class Start extends Command {
...
   }
+
+  async prepareNetwork(network: string, path: string) {
+    mkdirSync(path, {recursive: true})
+    if (network === "mesg-dev-chain") return // This is the default network, the engine will setup everything
+
+    const updateConfig = async (path: string, file: string, remote: string) => {
+      if (!existsSync(path)) mkdirSync(path, {recursive: true})
+      if (existsSync(join(path, file))) return
+      const data = await (await fetch(remote)).text()
```
Comment (NicolasMahe) [inline packages/cli/src/commands/daemon/start.ts]: catch not found error

File: packages/cli/src/commands/daemon/start.ts [hunk 3]
```diff
@@ -55,11 +62,13 @@ export default class Start extends Command {
       Action === 'start' &&
       from === `mesg/engine:${flags.version}`
     )
-    const network = await this.getOrCreateNetwork({name: flags.name})
-    await this.createEngineService(network, {
+    const configPath = join(flags.path, flags.network)
```
Comment (NicolasMahe) [inline packages/cli/src/commands/daemon/start.ts]: let's keep only one path for now

File: packages/cli/src/commands/daemon/start.ts [hunk 4]
```diff
@@ -32,6 +34,11 @@ export default class Start extends Command {
       default: 26656,
       required: true,
     }),
+    network: flags.string({
+      description: 'Name of the network to connect to',
+      default: 'mesg-dev-chain',
```
Comment (NicolasMahe) [inline packages/cli/src/commands/daemon/start.ts]: let's make this argument optional with empty string as default

**Review by NicolasMahe — CHANGES_REQUESTED** [review]

File: packages/cli/src/commands/daemon/start.ts (hunk 2 above)
Comment (antho1404) [reply packages/cli/src/commands/daemon/start.ts]: done

File: packages/cli/src/commands/daemon/start.ts (hunk 3 above)
Comment (antho1404) [reply packages/cli/src/commands/daemon/start.ts]: done

File: packages/cli/src/commands/daemon/start.ts (hunk 4 above)
Comment (antho1404) [reply packages/cli/src/commands/daemon/start.ts]: done

File: packages/cli/src/commands/daemon/start.ts [hunk 5]
```diff
@@ -83,4 +89,29 @@ export default class Start extends Command {
...
+
+  async prepareNetwork(network: string, path: string) {
+    mkdirSync(path, {recursive: true})
+    if (!network) return // This is the default network, the engine will setup everything
+
+    const updateConfig = async (path: string, file: string, remote: string) => {
+      if (!existsSync(path)) mkdirSync(path, {recursive: true})
+      if (existsSync(join(path, file))) return
+      const response = await fetch(remote)
+      if (response.status !== 200) throw new Error(`network ${network} doesn't exists`)
```
Comment (NicolasMahe) [inline packages/cli/src/commands/daemon/start.ts]:
```suggestion
      if (response.status !== 200) throw new Error(`network "${network}" doesn't exist`)
```

**Review by NicolasMahe — APPROVED** [review]

**Pull request merged**